sql/maintenance_utils.sql

```diff
@@ -62,61 +63,14 @@ CREATE OR REPLACE FUNCTION _timescaledb_functions.get_compressed_chunk_index_for
-CREATE OR REPLACE PROCEDURE @extschema@.recompress_chunk(chunk REGCLASS,
-                                             if_not_compressed BOOLEAN = true)
-AS $$
-DECLARE
-  status INT;
-  chunk_name TEXT[];
-  compressed_chunk_index REGCLASS;
-BEGIN
-
-    -- procedures with SET clause cannot execute transaction
-    -- control so we adjust search_path in procedure body
-    SET LOCAL search_path TO pg_catalog, pg_temp;
-
-    status := _timescaledb_functions.chunk_status(chunk);
 
-    -- Chunk names are in the internal catalog, but we only care about
-    -- the chunk name here.
-    -- status bits:
-    -- 1: compressed
-    -- 2: compressed unordered
-    -- 4: frozen
-    -- 8: compressed partial
-
-    chunk_name := parse_ident(chunk::text);
-    CASE
-    WHEN status = 0 THEN
-        RAISE EXCEPTION 'call compress_chunk instead of recompress_chunk';
-    WHEN status = 1 THEN
-        IF if_not_compressed THEN
-            RAISE NOTICE 'nothing to recompress in chunk "%"', chunk_name[array_upper(chunk_name,1)];
-            RETURN;
-        ELSE
-            RAISE EXCEPTION 'nothing to recompress in chunk "%"', chunk_name[array_upper(chunk_name,1)];
-        END IF;
-    WHEN status = 3 OR status = 9 OR status = 11 THEN
-        -- first check if there's an index. Might have to use a heuristic to determine if index usage would be efficient,
-        -- or if we'd better fall back to decompressing & recompressing entire chunk
-        SELECT _timescaledb_functions.get_compressed_chunk_index_for_recompression(chunk) INTO STRICT compressed_chunk_index;
-        IF compressed_chunk_index IS NOT NULL THEN
-            PERFORM _timescaledb_functions.recompress_chunk_segmentwise(chunk, if_not_compressed);
-        ELSE
-            PERFORM @extschema@.decompress_chunk(chunk);
-            COMMIT;
-            -- SET LOCAL is only active until end of transaction.
-            -- While we could use SET at the start of the function we do not
-            -- want to bleed out search_path to caller, so we do SET LOCAL
-            -- again after COMMIT
-            SET LOCAL search_path TO pg_catalog, pg_temp;
-            PERFORM @extschema@.compress_chunk(chunk, if_not_compressed);
-        END IF;
-    ELSE
-        RAISE EXCEPTION 'unexpected chunk status % in chunk "%"', status, chunk_name[array_upper(chunk_name,1)];
-    END CASE;
-END
-$$ LANGUAGE plpgsql;
+CREATE OR REPLACE PROCEDURE @extschema@.recompress_chunk(chunk REGCLASS, if_not_compressed BOOLEAN = true) LANGUAGE PLPGSQL AS $$
+BEGIN
+  IF current_setting('timescaledb.enable_deprecation_warnings', true)::bool THEN
+    RAISE WARNING 'procedure @extschema@.recompress_chunk(regclass,boolean) is deprecated and the functionality is now included in @extschema@.compress_chunk. this compatibility function will be removed in a future version.';
+  END IF;
+  PERFORM @extschema@.compress_chunk(chunk, if_not_compressed);
```

Comment:
```suggestion
  PERFORM @extschema@.compress_chunk(chunk, if_not_compressed, TRUE);
```
Shouldn't pass `recompress => true` here???